### ISSUE TITLE
Middleware to pass arbitrary data to handlers

### ISSUE THREAD

**Issue by sylc** [denoland/fresh#131]
I'm thinking that it would be nice if the middleware could pass some data down to the handler in a similar way of the handler passing data to the renderer, by providing an argument to ctx.handle(mydata).  
The data could then be retrieved by the handler using maybe `ctx.state`

example:

```ts
// _middleware.ts

import { MiddlewareHandlerContext } from "../server_deps.ts";

export async function handler(req: Request, ctx: MiddlewareHandlerContext) {
  const clientIDExisting = getCookies(req.headers)["clientID"];
  const clientID = clientIDExisting || crypto.randomUUID()

  // ** Pass data to handler 
  const resp = await ctx.handle({ clientID });

  if (!clientIDExisting) {
    setCookie(resp.headers, {
      name: "clientID",
      value: clientID,
      path: "/"
    });
  }
  
  return resp;

```
then in the handler, something like

```ts
export const handler: Handlers = {
  GET(_req, ctx) {
    return new Response(`clientID is ${ctx.state.clientID }`);
  },
};

```

My current workaround to achieve similiar feature is to add the data to the req object. i.e: `req['clientID'] = clientID`, which does not seem really clean

**Comment by lucacasonato** [comment]
Yeah, this seems like a great idea! Do you want to implement it?

This will also be awesome in combination with #136.

**Comment by lucacasonato** [comment]
Implemented by @sylc in #139.
